src/containers/apps/appDetails/deploy/Deployment.tsx

```diff
@@ -25,14 +26,27 @@ export default class Deployment extends ApiComponent<
       | undefined;
   }
 > {
-  constructor(props: any) {
+  initRepoInfo: RepoInfo;
+
+  constructor(props: AppDetailsTabProps) {
     super(props);
     this.state = {
       dummyVar: undefined,
       forceEditableCaptainDefinitionPath: false,
       updatedVersions: undefined,
       buildLogRecreationId: ""
```

Comment:
you might consider a `dirty` flag here that gets set when the `updateRepoInfo` callback is called and base the disable check on that

Reply:
Thanks @solomon23 but it's not part of the changes in this PR. Let's keep the scope of the PR limited so that it's easier to review and less chance for bugs to slip in.

Reply:
I was suggesting it for the pattern for this change.  This method he's working on checks if the repoInfo of the props is different than the one sent in when the app module was created ( which is problematic in react ).  Instead you could put in a check for when any of the repo fields change and mark it as dirty.  Once it's dirty you can enable the save button

Reply:
Ah I see. I thought you were commenting on log creation ID. 

I haven't gone through the changes yet, but your approach seems good if there's only one place that the repo info gets changed. Will look into this later when i go through the changes.